### PR TITLE
feat!: Add support for Svelte 4

### LINKED ISSUE #295: Svelte 4 support
### Describe the bug

`types/svelte.d.ts` uses deprecated types. I don't know if this library intends to keep supporting Svelte 3, but the type declaration should now be written like this:

```ts
declare module '~icons/*' {
  import { SvelteComponent } from 'svelte'
  import type { SVGAttributes } from 'svelte/elements'
  export default class extends SvelteComponent<SVGAttributes<EventTarget>> {}
}
```

Here is how it currently looks

image

- `svelte/internal` is not exposed anymore
- `SvelteComponentTyped` is deprecated
- `svelte` does not exist

I can open a merge request with these changes if breaking changes are accepted

### Reproduction

/

### System Info

```Shell
/
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by userquin**
closes #295

### CODE REVIEW

**Review by GauBen — APPROVED** [review]
Looks good to me, I've been using these definitions for a few time and so far so good

**Review by benmccann — APPROVED** [review]
lgtm

I wonder if we should also export `types/svelte` as an alias for whatever the newest version of svelte is

File: types/svelte.d.ts [hunk 1]
```diff
@@ -1,8 +1 @@
-declare module 'virtual:icons/*' {
-  export { SvelteComponentDev as default } from 'svelte/internal'
-}
-
-declare module '~icons/*' {
-  import { SvelteComponentTyped } from 'svelte'
-  export default class extends SvelteComponentTyped<svelte.JSX.IntrinsicElements['svg']> {}
-}
+import './svelte4.d.ts'
```
Comment (benmccann) [inline types/svelte.d.ts:1]: I would probably delete this file and update `./types/svelte` in `package.json` to point to `svelte4.d.ts` to simplify things

Comment (userquin) [reply types/svelte.d.ts:1]: node module resolution needs the file to be there

**Review by benmccann — APPROVED** [review]

**Review by userquin — COMMENTED** [review]
Ben remove the comment and I just send another comment (creating a new review)

**Review by benmccann — APPROVED** [review]
sorry. Your response to my earlier comment wasn't showing up, so I deleted my comment when I realized it was wrong

File: README.md [hunk 2]
```diff
@@ -498,8 +498,8 @@ Type Declarations
 
 For SvelteKit, in the `src/app.d.ts` file:
 
-```ts
-import 'unplugin-icons/types/svelte'
+```js
+/// <reference types="unplugin-icons/types/svelte" />
```
Comment (benmccann) [inline README.md:502]: I think this should be `import 'unplugin-icons/types/svelte';` here and in the example. I'm not much of a TypeScript expert, but my understanding is the the default `app.d.ts` in SvelteKit projects is written in such a way as to work with and expect `import` to be used: https://github.com/sveltejs/kit/issues/8276

Comment (userquin) [reply README.md]: done, also included global in `app.d.ts`

Comment (userquin) [reply README.md]: VSCode resolves virtuals properly, but missing types when using IntelliJ, maybe Svelte Plugin problem (the file is resolved with crtl + mouse click):

imagen

Comment (userquin) [reply README.md]: Adding a new `src/vite-env.d.ts` file with triple slash references works in VSCode and IntelliJ (Node 18, TypeScript 5).

I guess IntelliJ in monorepo with Vue and Svelte not working properly (screenshot using kit project), there are a lot of options in IntelliJ and TypeScript (Vue: Vola or TS, TS Lint, Svelte, TS itself...)

image

**Pull request merged**